### DISCUSSION TITLE
unpublish post through background job decoupled cms

### DISCUSSION THREAD

**Discussion by saleemhaq** [OrchardCMS/OrchardCore#10924] [Q&A]
Any help please help in below query:

Unpublish post through background job of any content type in decoupled cms.

**Comment by hishamco** [comment]
You could call `IContentManager.UnpublishAsync()` from within your background job

**Reply by saleemhaq** [reply]
> You could call `IContentManager.UnpublishAsync()` from within your background job

I implemented as you suggested please review the below screenshots, but still I facing the issue:

![image](https://user-images.githubusercontent.com/5202291/147408578-c7b20aa4-2629-4ff1-9548-fa7a60d23027.png)

The startup file look like:

![image](https://user-images.githubusercontent.com/5202291/147408598-b5f9d60e-79c5-44c3-b99d-ef269d02880d.png)

When I run the application is loaded not loading the application and in debug window of visual studio below are is showing:

![image](https://user-images.githubusercontent.com/5202291/147408660-89aafa42-a7d5-4131-9a29-a1f542c2aad9.png)

Anyone please help.

**Reply by jtkech** [reply]
Your bg task should be a scoped service, not a singleton

**Reply by saleemhaq** [reply]
> scoped service

Now the code is running but the content item not remove from admin side:

https://localhost:44300/Contents/ContentItems/43nh31sazqa2c56s4ymeq7be5m

![image](https://user-images.githubusercontent.com/5202291/147410043-688942c0-7be8-4063-8acf-5a39d197224a.png)

The page content is still showing:

![image](https://user-images.githubusercontent.com/5202291/147410060-4c4a9707-0317-4fb8-af7a-46663f3fab31.png)


Please advise.

**Reply by jtkech** [reply]
public async Task ...

    await _contentManager.RemoveAsync(...);

**Reply by saleemhaq** [reply]
> ```
> public async Task ...
> 
> await _contentManager.RemoveAsync(...);
> ```

Can i set the schedule time from admin side??? if yes so how i call then in code ?

![image](https://user-images.githubusercontent.com/5202291/147411546-ae743e22-77fd-4b24-a2c5-da9a66dcac66.png)

Further more how i get the updated content item in code side?

**Reply by saleemhaq** [reply]
@hishamco  @jtkech can u help me out in above my query please.

**Comment by hishamco** [comment]
@saleemhaq please make your question from the beginning to avoid unexpected answer, you said in the beginning that you want to unpublish the content item, late you are asking about remove them and finally you are asking about update the content

@jtkech was right for asked before, so @saleemhaq please make everything clear to save the time

Thanks

**Reply by saleemhaq** [reply]
> @saleemhaq please make your question from the beginning to avoid unexpected answer, you said in the beginning that you want to unpublish the content item, late you are asking about remove them and finally you are asking about update the content
> 
> @jtkech was right for asked before, so @saleemhaq please make everything clear to save the time
> 
> Thanks

Got your point @hishamco @jtkech can u help me out in above my query please.
